pyaptly/state_reader.py

```diff
@@ -175,4 +175,11 @@ def has_dependency(self, dependency):
             raise ValueError("Unknown dependency to resolve: %s" % str(dependency))
 
 
-state = SystemStateReader()
+_state_reader: SystemStateReader | None = None
+
+
+def get_state_reader():
```

Comment:
We could probably do without the `get_` prefix, I think it's just as readable / understandable if the function's just called `state_reader()`.